python/vowpalwabbit/sklearn_vw.py

```diff
@@ -560,6 +561,34 @@ class VWRegressor(VW, RegressorMixin):
     pass
 
 
+class VWMultiClassifier(OneVsRestClassifier, VW):
+    """Vowpal Wabbit Regressor model """
+
+    def __init__(self, **params):
+
+        super(VWClassifier, self).__init__(**params)
+
+    def predict(self, X):
+
+        """Fit underlying estimators.
+
+        Parameters
+        ----------
+        X : (sparse) array-like of shape (n_samples, n_features)
+            Data.
+        y : (sparse) array-like of shape (n_samples,) or (n_samples, n_classes)
+            Multi-class targets. An indicator matrix turns on multilabel
+            classification.
+
+        Returns
+        -------
+        self
+        """
+        pass
+
+
```

Comment:
Here, we have to pass an estimator to the `VWMultiClassifier` due to the `OneVsRestClassifier`.` OneVsRestClassifier` takes an estimator as an argument.So How should I override the decision function?

Reply:
the fit function of sklearn.
```julia

def fit(self, X, y):
--
  | """Fit underlying estimators.
  |  
  | Parameters
  | ----------
  | X : (sparse) array-like of shape (n_samples, n_features)
  | Data.
  |  
  | y : (sparse) array-like of shape (n_samples,) or (n_samples, n_classes)
  | Multi-class targets. An indicator matrix turns on multilabel
  | classification.
  |  
  | Returns
  | -------
  | self
  | """
  | # A sparse LabelBinarizer, with sparse_output=True, has been shown to
  | # outperform or match a dense label binarizer in all cases and has also
  | # resulted in less or equal memory consumption in the fit_ovr function
  | # overall.
  | self.label_binarizer_ = LabelBinarizer(sparse_output=True)
  | Y = self.label_binarizer_.fit_transform(y)
  | Y = Y.tocsc()
  | self.classes_ = self.label_binarizer_.classes_
  | columns = (col.toarray().ravel() for col in Y.T)
  | # In cases where individual estimators are very fast to train setting
  | # n_jobs > 1 in can results in slower performance due to the overhead
  | # of spawning threads.  See joblib issue #112.
  | self.estimators_ = Parallel(n_jobs=self.n_jobs)(delayed(_fit_binary)(
  | self.estimator, X, column, classes=[
  | "not %s" % self.label_binarizer_.classes_[i],
  | self.label_binarizer_.classes_[i]])
  | for i, column in enumerate(columns))
  |  
  | return self


```